sw/device/tests/rv_core_ibex_nmi_irq_test.c

```diff
@@ -0,0 +1,263 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <assert.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/math.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_alert_handler.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rv_core_ibex.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/dif/dif_rv_timer.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/alert_handler_testutils.h"
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/rand_testutils.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/FreeRTOSConfig.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_isrs.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "sw/device/lib/testing/autogen/isr_testutils.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+typedef void (*isr_handler)(void);
+
+static volatile isr_handler expected_isr_handler;
+static volatile dif_rv_core_ibex_nmi_state_t nmi_state;
+static volatile bool nmi_fired = false;
+
+static dif_pwrmgr_t pwrmgr;
+static dif_rv_core_ibex_t rv_core_ibex;
+static dif_aon_timer_t aon_timer;
+static aon_timer_isr_ctx_t aon_timer_ctx = {
+    .aon_timer = &aon_timer,
+    .plic_aon_timer_start_irq_id =
+        kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired,
+};
+
+static dif_rv_plic_t plic;
+static plic_isr_ctx_t plic_ctx = {
+    .rv_plic = &plic,
+    .hart_id = kTopEarlgreyPlicTargetIbex0,
+};
+
+static volatile dif_aon_timer_irq_t irq;
+static dif_alert_handler_t alert_handler;
+static alert_handler_isr_ctx_t alert_handler_ctx = {
+    .alert_handler = &alert_handler,
+    .plic_alert_handler_start_irq_id = kTopEarlgreyPlicIrqIdAlertHandlerClassa,
+    .expected_irq = kDifAlertHandlerIrqClassb,
+    .is_only_irq = false,
+};
+
+/**
+ * Program the alert handler to escalate on alerts upto phase 2 (i.e. reset)
+ * but at the phase 1 (i.e. wipe secrets)the nmi interrupt handler should clear
+ * the escalation.
+ */
+enum {
+  kEscalationPhase0Micros = 100 * 1000,  // 10 ms
+  kEscalationPhase2Micros = 100,         // 100 us
+};
```

Comment:
Why are these in an enum as opposed to just regular constants?

Reply:
See comment from [PR#14293](https://github.com/lowRISC/opentitan/pull/14293#discussion_r953195009). I believe @engdoreis is planning to add this to the style guide for future reference.

Reply:
Ah yes, I remember that. I think it'll still take me some time to get used to this construct. In this particular test we are not using them for switch statements nor array sizes.